ipv8/attestation/trustchain/community.py

```diff
@@ -157,31 +157,77 @@ def send_block_pair(self, block1, block2, address=None, ttl=2):
     def self_sign_block(self, block_type='unknown', transaction=None):
         self.sign_block(self.my_peer, block_type=block_type, transaction=transaction)
 
-    def sign_block(self, peer, public_key=EMPTY_PK, block_type='unknown', transaction=None, linked=None):
+    def create_source_no_counterparty(self, block_type='unknown', transaction=None):
+        """
+        Create a source block without any initial counterparty to sign.
+
+        :param block_type: The type of the block to be constructed, as a string
+        :param transaction: A string describing the interaction in this block
+        :return: None
+        """
+        self.sign_block(peer=None, public_key=ANY_COUNTERPARTY_PK, block_type=block_type, transaction=transaction)
+
+    def create_link(self, source, block_type='unknown', additional_info=None, public_key=None):
+        """
+        Create a Link Block to a source block
+
+        :param source: The source block which had no initial counterpary to sign
+        :param block_type: The type of the block to be constructed, as a string
+        :param additional_info: a dictionary with supplementary information concerning the transaction
+        :param public_key: The public key of the counterparty (usually of the source's owner)
+        :return: None
+        """
+        public_key = source.public_key if public_key is None else public_key
+
+        self.sign_block(self.my_peer,
+                        linked=source,
+                        public_key=public_key,
+                        block_type=block_type,
+                        additional_info=additional_info)
+
+    def sign_block(self,
```

Comment:
I prefer having these arguments on the same line actually (it is more consistent with the rest of our code).